src/utilities/indexer/queryFromIndexer.ts

```diff
@@ -0,0 +1,132 @@
+import { got } from 'got';
+
+import { configuration } from '../configuration';
+import { logger } from '../logger';
+import { sleep } from '../sleep';
+
+import { buildQuery } from './buildQuery';
+
+const { indexer } = configuration;
+
+const QUERY_INTERVAL_MS = 1000;
+export const QUERY_SIZE = 50;
+
+/** Example Query. */
+const queryBlocks = `
+  query {
+    blocks(orderBy: TIME_STAMP_DESC, first: 3) {
+    totalCount
+      nodes {
+        id
+        timeStamp
+        hash
+      }
+    }
+  }
+`;
+
+interface FetchedData {
+  data: Record<
+    string,
+    {
+      totalCount?: number;
+      nodes?: Array<Record<string, unknown>>;
+    }
+  >;
+}
+
+export async function queryFromIndexer(query: string = queryBlocks) {
+  logger.debug(
+    `Querying from GraphQL under ${indexer.graphqlEndpoint}, using this payload: ${query} `,
+  );
+  const { data } = await got
+    .post(indexer.graphqlEndpoint, {
+      json: {
+        query,
+      },
+    })
+    .json<FetchedData>();
+
+  const entities = Object.entries(data);
+
+  if (entities.length > 1) {
+    logger.error('Please, avoid multiple queries in a single request.');
+  }
+
+  const [name, { totalCount, nodes: matches }] = entities[0];
+
+  if (totalCount === undefined) {
+    logger.error(
+      'The query did not ask for total count. Please add field "totalCount" to your query.',
+    );
+  }
+  if (matches === undefined) {
+    logger.error(
+      'You need to include "nodes" as a field (with subfields) on your query to get matches.',
+    );
+  }
```

Comment:
You are logging these as errors, which implies that you consider them errors. If you throw errors at this point, then TypeScript will know that this function can only return defined values, and you won’t have to handle a weird case `if (!matches) {` below. I think this would be a good approach.

Reply:
I wanted to have a basic function that would not throw. I thought it will be useful for development during integration with other apps. (maybe even for the indexer docs)

Reply:
Succumbed here: [feat: intolerance while querying](https://github.com/galaniprojects/ctypehub/commit/032a855771025cd0c359adc5d6c233a1dc8fee48)